core/services/pipeline/orm.go

```diff
@@ -271,10 +271,41 @@ func (o *orm) InsertFinishedRun(run *Run, saveSuccessfulTaskRuns bool, qopts ...
 	return errors.Wrap(err, "InsertFinishedRun failed")
 }
 
+// DeleteRunsOlderThan deletes all pipeline_runs that have been finished for a certain threshold to free DB space
 func (o *orm) DeleteRunsOlderThan(ctx context.Context, threshold time.Duration) error {
 	q := o.q.WithOpts(pg.WithParentCtx(ctx))
-	err := q.ExecQ(`DELETE FROM pipeline_runs WHERE finished_at < $1`, time.Now().Add(-threshold))
-	return errors.Wrap(err, "DeleteRunsOlderThan failed")
+
+	err := pg.Batch(func(_, limit uint) (count uint, err error) {
+		result, cancel, err := q.ExecQIter(`
```

Comment:
You may consider overriding the context here with a longer timeout also

Reply:
Oh yeah - can just move 277 down inside here, right?

Reply:
Or is that already being applied per-batch call, even though only one instance is created?

Reply:
My understanding is that it should apply the custom timeout even though there's a single instance per batch call @jmank88.